crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,349 @@
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            BluesnapPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::foreign_try_from((
+                item.response.card_transaction_type,
+                item.response.processing_info.processing_status,
+            ))?,
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    item.response.transaction_id,
+                ),
+                redirection_data: None,
+                mandate_reference: None,
+                connector_metadata: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Serialize)]
+pub struct BluesnapRefundRequest {
+    amount: Option<i64>,
+    reason: Option<String>,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for BluesnapRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            reason: item.request.reason.clone(),
+            amount: Some(item.request.refund_amount),
+        })
+    }
+}
+
+#[derive(Default, Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RefundResponse {
+    refund_transaction_id: i32,
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::RSync, BluesnapPaymentsResponse>>
+    for types::RefundsRouterData<api::RSync>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::RSync, BluesnapPaymentsResponse>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: Ok(types::RefundsResponseData {
+                connector_refund_id: item.response.transaction_id.clone(),
+                refund_status: enums::RefundStatus::foreign_try_from((
+                    item.response.card_transaction_type,
+                    item.response.processing_info.processing_status,
+                ))?,
+            }),
+            ..item.data
+        })
+    }
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, RefundResponse>>
+    for types::RefundsRouterData<api::Execute>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Execute, RefundResponse>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: Ok(types::RefundsResponseData {
+                connector_refund_id: item.response.refund_transaction_id.to_string(),
+                refund_status: enums::RefundStatus::Pending,
```

Comment:
Why is this always `Pending`?

Reply:
As the connector is not sending any status, we cannot directly make success it can be pending on bank's server or in any other flow it might be stuck so we make it pending and do a `Rsync` over it